sonaranalyzer-dotnet/src/SonarAnalyzer.VisualBasic/Rules/BooleanLiteralUnnecessary.cs

```diff
@@ -0,0 +1,137 @@
+
+namespace SonarAnalyzer.Rules.VisualBasic
+{
+    [DiagnosticAnalyzer(LanguageNames.VisualBasic)]
+    [Rule(DiagnosticId)]
+    public sealed class BooleanLiteralUnnecessary : BooleanLiteralUnnecessaryBase<BinaryExpressionSyntax, SyntaxKind>
+    {
+        private static readonly DiagnosticDescriptor rule =
+            DiagnosticDescriptorBuilder.GetDescriptor(DiagnosticId, MessageFormat, RspecStrings.ResourceManager);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(rule);
+        protected override DiagnosticDescriptor Rule { get; } = rule;
+        protected override SyntaxKind TrueLiteral { get; } = SyntaxKind.TrueLiteralExpression;
+        protected override SyntaxKind FalseLiteral { get; } = SyntaxKind.FalseLiteralExpression;
+
+        protected override SyntaxNode Left(BinaryExpressionSyntax binaryExpression) => binaryExpression.Left.RemoveParentheses();
+
+        protected override SyntaxNode Right(BinaryExpressionSyntax binaryExpression) => binaryExpression.Right.RemoveParentheses();
+
+        protected override SyntaxToken OperatorToken(BinaryExpressionSyntax binaryExpression) => binaryExpression.OperatorToken;
+
+        protected override bool IsKind(SyntaxNode syntaxNode, SyntaxKind syntaxKind) => syntaxNode.IsKind(syntaxKind);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckLogicalNot,
+                SyntaxKind.NotExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckAndExpression,
+                SyntaxKind.AndAlsoExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckOrExpression,
+                SyntaxKind.OrElseExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckEquals,
+                SyntaxKind.EqualsExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckNotEquals,
+                SyntaxKind.NotEqualsExpression);
+
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                CheckConditional,
+                SyntaxKind.TernaryConditionalExpression);
+        }
+
+        private void CheckLogicalNot(SyntaxNodeAnalysisContext context)
+        {
+            var logicalNot = (UnaryExpressionSyntax)context.Node;
+            var logicalNotOperand = logicalNot.Operand.RemoveParentheses();
+            if (IsBooleanLiteral(logicalNotOperand))
+            {
+                context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, logicalNot.Operand.GetLocation()));
+            }
+            bool IsBooleanLiteral(SyntaxNode node) =>
+                node.IsKind(SyntaxKind.TrueLiteralExpression) || node.IsKind(SyntaxKind.FalseLiteralExpression);
+        }
+
+        private void CheckConditional(SyntaxNodeAnalysisContext context)
```

Comment:
I've left the conditional as "native" because:
- when trying to move it to the base class, the C# code fixer was failing and I didn't manage to find out why (and I didn't want to stay blocked)
- I'd have to add a 3rd generic parameter to the base class implementation (for `TernaryConditionalExpressionSyntax` (VB) / `ConditionalExpressionSyntax` (C#) ) which we generally want to avoid